Compilation error due to libpng

While trying to compile Choreonoid on Arch Linux, I get the following compilation error:

```txt
[  7%] Building CXX object src/Util/CMakeFiles/CnoidUtil.dir/ImageIO.cpp.o
In file included from /usr/include/pngconf.h:50:0,
                 from /usr/include/png.h:344,
                 from /home/user/dev/choreonoid/src/Util/ImageIO.cpp:11:
/home/user/dev/choreonoid/src/Util/ImageIO.cpp: In function ‘void {anonymous}::savePNG(const cnoid::Image&, const string&, bool)’:
/home/user/dev/choreonoid/src/Util/ImageIO.cpp:186:22: error: invalid use of incomplete type ‘png_struct {aka struct png_struct_def}’
     if(setjmp(png_ptr->jmpbuf)){
                      ^
In file included from /home/user/dev/choreonoid/src/Util/ImageIO.cpp:11:0:
/usr/include/png.h:459:16: note: forward declaration of ‘png_struct {aka struct png_struct_def}’
 typedef struct png_struct_def png_struct;
                ^
```

This feature was recently implemented in https://github.com/s-nakaoka/choreonoid/commit/4036b81861884ffe7ef3cbd8dbf3d558a6c51f32.

The version of libpng is 1.6.20 on my system. I've got a patch on the way.

--- a/src/Util/ImageIO.cpp
+++ b/src/Util/ImageIO.cpp
@@ -183,7 +183,7 @@ void savePNG(const Image& image, const std::string& filename, bool isUpsideDown)
         throwSaveException(filename, strerror(errno));
     }
     
-    if(setjmp(png_ptr->jmpbuf)){
+    if(setjmp(png_jmpbuf(png_ptr))){
         png_destroy_write_struct(&png_ptr, (png_infopp)NULL);
         fclose(fp);
         throwSaveException(filename, "Internal error.");

